src/main/java/se/heinrisch/talkclient/TalkClient.java

```diff
@@ -167,9 +180,11 @@ public void onResult(DataItemBuffer dataItems) {
                     final DataMapItem dataMapItem = DataMapItem.fromDataItem(dataItems.get(i));
                     if (dataMapItem.getUri().getPath().equals(path)) {
                         mTalkDataAdapter.onRecieveResult(dataMapItem.getDataMap());
-                        continue;
+                        dataItems.release();
+                        return;
```

Comment:
It seems I did something wrong when handling the result because the above works as expected.
When testing I've had mTalkDataAdapter.onRecieveResult(new DataMap()) commented out though. As long as it is handled by the "event handler" it is no problem though. So please merge this.